workshops/git_101/02-clone-commit-push-merge.md

```diff
@@ -0,0 +1,161 @@
+
+```shell
+Cloning into 'HowWeEnable'...
+remote: Enumerating objects: 283, done.
+remote: Counting objects: 100% (283/283), done.
+remote: Compressing objects: 100% (173/173), done.
+remote: Total 283 (delta 77), reused 276 (delta 74), pack-reused 0
+Receiving objects: 100% (283/283), 81.03 KiB | 1.72 MiB/s, done.
+Resolving deltas: 100% (77/77), done.
+cd HowWeEnable
+tree
+.
+├── LICENSE
+├── README.md
+└── workshops
+    └── how_we_enable
+        ├── 01-Getting_Started.md
+        ├── 02-Logistics.md
+        ├── 03-Strategy.md
+        └── 04-Practice_Scenarios.md
+```
+
+## Branching
+
+It is a best practice to not commit code directly to the main (legacy master) branch. In order to void this we create a feature branch to work from with the `git branch` or `git checkout` command.
+
+To create a branch and check it out (ie move to it as your working branch) you can do this with a single command `git checkout -b <branch-name>`.
+
+```shell
+cd HowWeEnable
+git checkout -b test
+```
+
+Output:
+
+```shell
+Switched to a new branch 'test'
+```
+
+Now any changes you make are in a separate branch from main called test.
+
+## Commiting Code
+
+1. In the HowWeEnable directory create a new file and look at the status of it in git.
+
+    ```shell
+    touch mynewfile.txt
+    git status
+    ```
+
+    Output of git status:
+
+    ```shell
+    Untracked files:
+    (use "git add <file>..." to include in what will be committed)
+        mynewfile.txt
+    ```
+
+2. Now let's git add so the file is tracked by git going forward, and git commit so the new file is ready to be pushed to the remote repository.
+
+    ```shell
+    git add mynewfile.txt
+    git status
+    ```
+
+    Output of git status:
+
+    ```shell
+    Changes to be committed:
+    (use "git restore --staged <file>..." to unstage)
+            new file:   _config.yml
+    ```
+
+3. Finally lets commit the code:
+
+    ```shell
+    git commit -m 'adding newfile.txt'
+    ```
+
+    Output:
+
+    ```shell
+    [main 60cdc2b] adding newfile.txt
+     1 file changed, 0 insertions(+), 0 deletions(-)
+    create mode 100644 newfile.txt
+    ```
+
+## Pushing Commit to Repository Server
+
+Just like it sounds, `git push` pushes the commits to the central repository. A best practice is to specify the repo you are pushing to (usually origin) and the branch for this example we will use test. `Note:` Generally upstream repositories will not allow pushing directly to main, so you will always need to either fork the repository, or create a branch.
+
+```shell
+git push origin test
+```
+
+Output:
+
+```shell
+Enumerating objects: 1, done.
+Counting objects: 100% (1/1), done.
+Delta compression using up to 8 threads
+Compressing objects: 100% (1/1), done.
+Writing objects: 100% (1/1), 1018 bytes | 339.00 KiB/s, done.
+Total 9 (delta 5), reused 0 (delta 0), pack-reused 0
+remote: Resolving deltas: 100% (1/1), completed with 1 local objects.
+To github.com:redhat-adsa/HowWeEnable.git
```

Comment:
Did you want to specify `yourgithubusername` instead of `redhat-adsa`

Reply:
```suggestion
To github.com:yourusername/HowWeEnable.git
```